pkg/zfs/zfs_util.go

```diff
@@ -353,6 +378,51 @@ func DestroyVolume(vol *apis.ZFSVolume) error {
 	return nil
 }
 
+// CreateSnapshot creates the zfs volume snapshot
+func CreateSnapshot(snap *apis.ZFSSnapshot) error {
+
+	volume := snap.Labels[ZFSVolKey]
+	snapDataset := snap.Spec.PoolName + "/" + volume + "@" + snap.Name
+
+	if err := getVolume(snapDataset); err == nil {
+		logrus.Infof("snapshot already there %s", snapDataset)
+		// snapshot already there just return
+		return nil
+	}
```

Comment:
How about command itself failing for some other reason giving a non-zero ? Will this log message be correct at that time?

Reply:
if command fails, we will not get this log.